music21/musicxml/m21ToXml.py

```diff
@@ -1735,6 +1736,66 @@ def parseFlatScore(self):
         pp.parse()
         self.partExporterList.append(pp)
 
+    def prePartProcess(self):
+        '''
+        Scans the parts/measures/voices in each joinable group to compute
+        the voiceNumberOffset needed in each part.  Parts that aren't in a
+        joinable group get offset 0.
+
+        Called automatically by .parse()
+        '''
+        partExporterForPart: t.Dict[stream.Part, PartExporter] = {}
+
+        # Compute partExp.staffGroup and partExp.maxNumVoices
+        for partExp in self.partExporterList:
+            partExporterForPart[partExp.stream] = partExp
+
+            joinableGroup = None
+            for sg in self.groupsToJoin:
+                if partExp.stream in sg:
+                    joinableGroup = sg
+                    break
+
+            partExp.staffGroup = joinableGroup
+            if partExp.staffGroup is None:
+                partExp.maxVoiceNum = 0
+                continue
+
+            # part is in a staffGroup, so we need to find the maximum number
+            # of voices used in any measure in this part.  We also find the
+            # maximum "low enough to not be considered a memory location"
+            # voice id.  The maxVoiceNum for a part is the higher of
+            # maxNumVoices-1 and maxNonMemLocationVoiceId.
+            maxNumVoices = 0
+            maxNonMemLocationVoiceId = 0
+            for measure in partExp.stream[stream.Measure]:
+                voices = tuple(measure[stream.Voice])
```

Comment:
```suggestion
                voices = measure[stream.Voice]
```

Reply:
I'll see if this works; I was assuming that measure[stream.Voice] would return an iterator, so I would need to make an iterable of it in order to call len().

Reply:
Well, isn't that lovely!  `StreamIterator.__len__()` for the win.